src/tests/end-to-end/tests/details-view/fast-pass-report.test.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { FeatureFlags } from 'common/feature-flags';
+import { Download } from 'playwright';
+import {
+    detailsViewSelectors,
+    fastPassReportSelectors,
+    tabStopsSelectors,
+} from 'tests/end-to-end/common/element-identifiers/details-view-selectors';
+import { BackgroundPage } from 'tests/end-to-end/common/page-controllers/background-page';
+import { HtmlReportPage } from 'tests/end-to-end/common/page-controllers/html-report-page';
+import { getTestResourceUrl } from 'tests/end-to-end/common/test-resources';
+import { Browser } from '../../common/browser';
+import { launchBrowser } from '../../common/browser-factory';
+import { DetailsViewPage } from '../../common/page-controllers/details-view-page';
+import { TargetPage } from '../../common/page-controllers/target-page';
+import { scanForAccessibilityIssues } from '../../common/scan-for-accessibility-issues';
+
+const reportDownloadUrl = getTestResourceUrl('fast-pass-report.html');
+
+describe('Details View -> FastPass -> Report', () => {
+    let browser: Browser;
+    let targetPage: TargetPage;
+    let detailsViewPage: DetailsViewPage;
+    let backgroundPage: BackgroundPage;
+    let reportPage: HtmlReportPage;
+    const reportSaveAsFilePath = 'src/tests/end-to-end/test-resources/fast-pass-report.html';
+    let reportDownload: Download;
+
+    beforeAll(async () => {
+        browser = await launchBrowser({
+            suppressFirstTimeDialog: true,
+            addExtraPermissionsToManifest: 'all-origins',
+        });
+        targetPage = await browser.newTargetPage();
+        await browser.newPopupPage(targetPage); // Required for the details view to register as having permissions/being open
+        backgroundPage = await browser.backgroundPage();
+        await backgroundPage.enableFeatureFlag(FeatureFlags.newTabStopsDetailsView);
+        detailsViewPage = await browser.newDetailsViewPage(targetPage);
+        await detailsViewPage.waitForSelector(detailsViewSelectors.automatedChecksResultSection);
+        await openTabStopsPage(detailsViewPage);
+        await addTabStopsFailure(detailsViewPage);
+        await detailsViewPage.waitForAndClickSelector(detailsViewSelectors.exportReportButton);
+        await detailsViewPage.waitForSelector(detailsViewSelectors.singleExportToHtmlButton);
+        reportDownload = await detailsViewPage.downloadExportReport(
+            detailsViewSelectors.singleExportToHtmlButton,
+            reportSaveAsFilePath,
+        );
+        reportPage = await browser.newHtmlReportPage(reportDownloadUrl);
+    });
+
+    afterAll(async () => {
+        // delete GUID file auto-generated by the download
+        await detailsViewPage.deleteDownloadedFile(reportDownload);
```

Comment:
It would be good to make sure that the `browser?.close()` step still runs if either/both of the `delete` steps fail. Particularly, `afterAll` still runs if the `beforeAll` fails; if any setup step fails before the download happens, we want the browser to still get cleaned up